lib/helpers.ts

```diff
@@ -0,0 +1,21 @@
+export interface Scenarios {
+  onMatch: (user: any) => any,
+  onCreateUser: (user: any) => any,
+  onAuthnFailed: (failure: any) => any,
+  onError: (error: any) => any,
+}
+
+export function createResponseHandler (scenarios: Scenarios): any {
+  return function (error: Error, user: any, infoOrError: any, status: number) {
+    if (error) {
+      return scenarios.onError(error)
+    }
+    if (user) {
+      if (user._isNewUser) {
```

Comment:
`infoOrError`-parameter would contain the original `TranslatedResponseBody`-object. We could inspect the presence of `attributes` field on that object instead of polluting `_isNewUser` on an object owned by the application.